lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -55,14 +55,14 @@ pub(crate) fn delete_points(
 pub(crate) fn set_payload(
     segments: &SegmentHolder,
     op_num: SeqNumberType,
-    payload: &HashMap<PayloadKeyType, PayloadInterface>,
+    payload: &serde_json::Value,
```

Comment:
I added a type alias in `types.rs` but I haven't used it yet because the scope of the refactoring wasn't clear at the beginning. In later commits I will wrap the json value and I'll provide some methods to access nested values.